tensorboard/data/server/cli.rs

```diff
@@ -254,6 +272,23 @@ fn write_port_file(path: &Path, port: u16) -> std::io::Result<()> {
     Ok(())
 }
 
+/// Writes error to the given file, or to stderr as a fallback.
+fn write_startup_error(path: Option<&Path>, error: dynamic_logdir::Error) {
+    let write_to_file = |path: &Path| -> std::io::Result<()> {
+        let mut f = File::create(path)?;
+        writeln!(f, "{}", error)?;
+        f.sync_all()?;
+        Ok(())
+    };
+    let should_write_to_stderr = match path {
+        None => true,
+        Some(p) => write_to_file(p).is_err(),
```

Comment:
Is it worth surfacing the error writing to the error file in some way?

Reply:
LGTM.  I just figured it was a bit weird to fail to fail to do what was instructed and not have any way for the user to assess why.  Also I guess my thought was that `--error-file` was more about being able to easily extract the critical error w/out including potentially unrelated stuff from stderr rather than about keeping stderr free from a critical error that is semi-expected, but I can see that it sort of serves both purposes.